crates/wash/src/lib/start/github/api.rs

```diff
@@ -12,62 +13,118 @@ type DateTimeUtc = DateTime<Utc>;
+}
+
+pub async fn new_minor_tag_pattern_releases_after(
+    owner: &str,
+    repo: &str,
+    after_version: &Version,
+    tag_pattern: Option<&str>,
+) -> Result<Vec<GitHubRelease>, Error> {
+    new_releases_after(owner, repo, tag_pattern, after_version, VersionMatch::Minor).await
+}
+
+enum VersionMatch {
+    Minor,
+    Patch,
+}
+impl VersionMatch {
+    fn as_str(&self) -> &str {
+        match self {
+            VersionMatch::Minor => "^",
+            VersionMatch::Patch => "~",
+        }
+    }
+}
+
+/// Get a full list of releases from the GitHub repository that are newer than the provided version.
+/// Optionally, a tag pattern can be provided to filter the releases. If this is not provided, only the main releases
+/// are considered.
+async fn new_releases_after(
+    owner: &str,
+    repo: &str,
+    tag_pattern: Option<&str>,
+    after_version: &Version,
+    comparator: VersionMatch,
+) -> Result<Vec<GitHubRelease>, Error> {
+    let releases = get_sorted_releases(owner, repo, after_version, tag_pattern).await?;
+    let op = comparator.as_str();
     let main_releases = releases
         .into_iter()
-        .filter(|release| match &release.get_main_artifact_release() {
-            Some(version) => {
-                after_version.major == version.major && after_version.minor == version.minor
-            }
-            None => false,
+        .filter(|release| match &release.get_x_y_z_version() {
+            Ok(_) => release.satisfies_constraint(&format!("{op}{v}, >{v}", v = after_version)),
+            _ => false,
         })
         .collect::<Vec<GitHubRelease>>();
     Ok(main_releases)
 }
 
-/// Returns the latest patch version of the provided version.
-pub async fn new_patch_version_of_after_string(
+/// Returns the version of a release if it is semver compatible (^) with the provided version. Optionally, a tag pattern
+/// can be provided to filter the releases. If this is not provided, only the main releases are considered.
+pub async fn new_minor_version_compatible_with_version_string(
     owner: &str,
     repo: &str,
-    after_version: &str,
-) -> Result<Option<Version>, Error> {
-    let after_version = after_version.strip_prefix('v').unwrap_or(after_version);
-    let after_version = match Version::parse(after_version) {
-        Ok(v) => v,
-        Err(_) => return Ok(None),
-    };
-    match new_patch_releases_after(owner, repo, &after_version).await {
-        Ok(patches) => match patches.first() {
-            Some(patch) => Ok(patch.get_main_artifact_release()),
-            None => Ok(None),
-        },
-        _ => Ok(None),
-    }
+    tag_pattern: Option<&str>,
+    version_string: &str,
+) -> Result<Version, Error> {
+    let version_string = Version::parse(version_string).expect("failed to parse version");
+    let releases = new_minor_tag_pattern_releases_after(owner, repo, &version_string, tag_pattern)
```

Comment:
```suggestion
    let version_string = version_string.strip_prefix('v').unwrap_or(version_string);
    let version = Version::parse(version_string).expect("failed to parse version");
```

There's an error from I think not stripping https://github.com/wasmCloud/wasmCloud/actions/runs/14256074065/job/39959172214?pr=4325

Reply:
Ok, yeah that's good. I've added another couple of integration tests to cover this.

https://github.com/lachieh/wasmCloud/blob/5b33c4d0e7e456398f31067b485761d32690ab8c/crates/wash/tests/github_fetch_test.rs#L233-L276